File: app/src/main/java/com/example/myanimedata/api/AnimeDetail.java
package com.example.myanimedata.api;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class AnimeDetail {
    @SerializedName("mal_id")
    private final int id;

    @SerializedName("title")
    private final String title;

    @SerializedName("title_japanese")
    private final String titleJp;

    @SerializedName("type")
    private final String type;

    @SerializedName("score")
    private final double score;

    @SerializedName("episodes")
    private final int episodes;

    @SerializedName("synopsis")
    private final String synopsis;

    @SerializedName("background")
    private final String background;

    @SerializedName("images")
    private final ImageResult imageResult;

    @SerializedName("trailer")
    private final TrailerResult trailerResult;

    @SerializedName("genres")
    private final List<GenreResult> genreResults = new ArrayList<>();

    public int getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public double getScore() {
        return score;
    }

    public int getEpisodes() {
        return episodes;
    }

    public String getSynopsis() {
        return synopsis;
    }

    public String getTitleJp() {
        return titleJp;
    }

    public String getType() {
        return type;
    }

    public ImageResult getImageResult() {
        return imageResult;
    }

    public TrailerResult getTrailerResult() {
        return trailerResult;
    }

    public String getBackground() {
        return background;
    }

    public List<GenreResult> getGenreResults() {
        return genreResults;
    }

    public AnimeDetail(int id, String title, String titleJp, String type, double score, int episodes,
                       String synopsis, String background, ImageResult imageResult,
                       TrailerResult trailerResult) {
        this.id = id;
        this.title = title;
        this.titleJp = titleJp;
        this.type = type;
        this.score = score;
        this.episodes = episodes;
        this.synopsis = synopsis;
        this.background = background;
        this.imageResult = imageResult;
        this.trailerResult = trailerResult;
    }
}

File: app/src/main/java/com/example/myanimedata/api/AnimeDetail.kt
package com.example.myanimedata.api

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class AnimeDetail(
    @field:SerializedName("mal_id") val id: Int? = 0,
    @field:SerializedName("title") val title: String? = null,
    @field:SerializedName("type") val type: String? = null,
    @field:SerializedName("score") val score: Double? = 0.0,
    @field:SerializedName("episodes") val episodes: Int? = null,
    @field:SerializedName("synopsis") val synopsis: String? = null,
    @field:SerializedName("background") val background: String? = null,
    @field:SerializedName("images") val imageResult: ImageResult? = null,
    @field:SerializedName("genres") val genreResults: ArrayList<GenreResult>? = null
) : Parcelable